src/epoch.rs

```diff
@@ -2226,6 +2226,82 @@ impl Epoch {
         me
     }
 
+    /// Builds an Epoch from given `wk` week counter into the desired Time scale.
+    /// `ns` is the amount of nanoseconds into that week from closest Sunday Midnight.
+    /// This is usually how GNSS receivers describe a GNSS time scale epoch
+    #[cfg(feature = "python")]
+    #[staticmethod]
+    pub fn from_timeofweek(wk: u32, ns: u64, ts: TimeScale) -> Self {
+        let week = Duration::from_seconds(wk as f64 * SECONDS_PER_DAY * 7.0);
+        Self::from_duration(week + (ns as f64) * Unit::Nanosecond, ts)
+    }
+
+    #[cfg(feature = "python")]
+    #[staticmethod]
+    pub fn from_timeofweek_utc(wk: u32, ns: u64) -> Self {
+        let week = Duration::from_seconds(wk as f64 * SECONDS_PER_DAY * 7.0);
+        Self::from_utc_duration(week + (ns as f64) * Unit::Nanosecond)
+    }
+
+    /// Converts to "time of week" (`wk, `tow`),
+    /// which is usually how GNSS receivers describe a timestamp.
+    /// `wk` is a rolling week conter into that time scale,
+    /// `tow` is the number of seconds since closest Sunday midnight into that week.
+    pub fn to_timeofweek_utc(&self) -> (u32, u64) {
+        // fractionnal days in this time scale
+        let days = match self.time_scale {
+            TimeScale::GPST => self.to_gpst_days(),
+            TimeScale::GST => self.to_gst_days(),
+            TimeScale::BDT => self.to_bdt_days(),
+            TimeScale::TT => self.to_tt_days(),
+            TimeScale::TAI => self.to_tai_days(),
+            TimeScale::UTC => self.to_utc_days(),
+            TimeScale::TDB => self.to_tdb_seconds() / SECONDS_PER_DAY,
+            TimeScale::ET => self.to_et_seconds() / SECONDS_PER_DAY,
+        };
+        // wk: rolling week counter into timescale
+        let wk = (days / 7.0) as u32;
```

Comment:
Should this be `floor()`?

Also, just to make sure I understand, the reason this works is because the reference epoch is on 01 January 1900 TAI and that day happens to be a Monday, is that right?

Reply:
> Should this be `floor()`?

that is true

> Also, just to make sure I understand, the reason this works is because the reference epoch is on 01 January 1900 TAI and that day happens to be a Monday, is that right?

that's correct